app/gen-server/migration/1756918816559-ServiceAccounts.ts

```diff
@@ -0,0 +1,97 @@
+import { MigrationInterface, QueryRunner, Table, TableColumn, TableForeignKey, TableIndex } from "typeorm";
+import * as sqlUtils from "app/gen-server/sqlUtils";
+
+export class ServiceAccounts1756918816559 implements MigrationInterface {
+
+  public async up(queryRunner: QueryRunner): Promise<any> {
+    const dbType = queryRunner.connection.driver.options.type;
+    const datetime = sqlUtils.datetime(dbType);
+
+    await queryRunner.addColumns('users', [
+      new TableColumn({
+        name: 'type',
+        type: 'varchar',
+        isNullable: false,
+        default: "'login'",
+      }),
+    ]);
+    await queryRunner.createTable(
+      new Table({
+        name: 'service_accounts',
+        columns: [
+          {
+            name: 'id',
+            type: 'integer',
+            isGenerated: true,
+            generationStrategy: 'increment',
+            isPrimary: true,
+          },
+          {
+            name: 'owner_id',
+            type: 'integer',
+          },
+          {
+            name: 'service_user_id',
+            type: 'integer',
+            isNullable: false,
+            isUnique: true,
+          },
+          {
+            name: 'label',
+            type: 'varchar',
+            isNullable: false,
+            default: "''",
```

Comment:
What's the rationale for using a non-nullable-with-empty-default varchar here instead of a nullable varchar as seems to be the common practice in `grist-core`?

I mean I get that nullable varchar annoyingly allows two representations for unset values (`NULL` and `''`) but consistency with the rest of the schema feels more important.